src/consensus.rs

```diff
@@ -90,6 +93,15 @@ impl Consensus {
                     eprintln!("{:?}", err)
                 }
             }
+            res = crate::wallet::run(
+                wallet_lock,
+                broadcast_channel_sender.clone(),
+                broadcast_channel_sender.subscribe()
+            ) => {
+                if let Err(err) = res {
+                    eprintln!("{:?}", err)
+                }
+            }
```

Comment:
What is the purpose of wallet run function? it seem to be an initialization, but I'm not clear why it's done this way.

Reply:
I think these are positive changes. Happy to switch to Mutex as well as above. Rationale for an independent class is that if wallet can listen on OnNewBlock or OnChainReorganization and then fetch the slips or update the spendable ones instead of being manually triggered by the blockchain class that would be an improvement.

Right now this class mostly exists as a place to store the Keypair so that we can get the publickey and privatekey into the mempool for transaction / block creation and signing in order to get blocks and transaction signature creation and validation worked on.